### ISSUE TITLE
Some items not looting after upgrade from 1.5 to 1.6

### ISSUE THREAD

**Issue by jtolmer** [SteveTownsend/SmartHarvestSE#104]
After upgrading my save from 1.5 to 1.6 some things aren't looting. For example, an enchanted mace in a weapon rack in Ashfallow Citadel and unread books on Seeker corpses in Apocrypha didn't loot. In both cases, if I downgrade back to 1.5, these items loot when loading the save. I haven't fully isolated the behavior to clear steps. That is, I had other unread books on a bookshelf that did get looted and armor looting off corpses, so some things are working. If there are specific things I can do to help track down the issue, please let me know.

I did do a short run in Ashfallow Citadel to where the weapon rack mace is located while running using both 1.5 and 1.6. Doing a diff of the generated SHSE logs doesn't reveal anything that looks relevant to me. The 1.6 log is 8 lines longer, if that means anything. Here's the bottom of that log:

```J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1300): [MESSAGE] {6044} (2020-07-04 12:23:47 PM) Plugin now in sync - Game Data load complete!
J:\GitHub\SmartHarvestSE\src\main.cpp(46): [ERROR] {6044} (2020-07-04 12:23:47 PM) Initialized SearchTask: Looting unavailable
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(57): [WARNING] {456} (2020-07-04 12:23:47 PM) VM Good to Go = 1 for request 58, expected request 0
J:\GitHub\SmartHarvestSE\src\Collections\CollectionManager.cpp(495): [MESSAGE] {16004} (2020-07-04 12:23:49 PM) Collections are disabled
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1227): [MESSAGE] {16004} (2020-07-04 12:23:49 PM) Restrictions reset, new/loaded game = true
J:\GitHub\SmartHarvestSE\src\VM\papyrus.cpp(237): [MESSAGE] {456} (2020-07-04 12:23:49 PM) Reference Search enabled
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(763): [MESSAGE] {15600} (2020-07-04 12:23:49 PM) Starting SHSE Worker Thread
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(889): [MESSAGE] {15600} (2020-07-04 12:23:50 PM) UI/controls now good-to-go
J:\GitHub\SmartHarvestSE\src\WorldState\LocationTracker.cpp(215): [WARNING] {15600} (2020-07-04 12:23:50 PM) Player cell not yet set up
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(100): [MESSAGE] {15600} (2020-07-04 12:26:28 PM) UI/controls no longer good-to-go
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(889): [MESSAGE] {15600} (2020-07-04 12:26:39 PM) UI/controls now good-to-go
J:\GitHub\SmartHarvestSE\src\WorldState\LocationTracker.cpp(215): [WARNING] {15600} (2020-07-04 12:26:39 PM) Player cell not yet set up
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(100): [MESSAGE] {15600} (2020-07-04 12:27:00 PM) UI/controls no longer good-to-go
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(89): [WARNING] {15600} (2020-07-04 12:27:41 PM) plugin UI good-to-go 0 (UI 0, controls 1) does not match VM UI good-to-go 1 - trust VM
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(889): [MESSAGE] {15600} (2020-07-04 12:27:41 PM) UI/controls now good-to-go
J:\GitHub\SmartHarvestSE\src\WorldState\LocationTracker.cpp(215): [WARNING] {15600} (2020-07-04 12:27:41 PM) Player cell not yet set up
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(89): [WARNING] {15600} (2020-07-04 12:27:46 PM) plugin UI good-to-go 1 (UI 1, controls 1) does not match VM UI good-to-go 0 - trust VM
J:\GitHub\SmartHarvestSE\src\VM\UIState.cpp(100): [MESSAGE] {15600} (2020-07-04 12:27:46 PM) UI/controls no longer good-to-go
```

**Comment by jtolmer** [comment]
I don't think this matters, but in case... When 1.5 came out, I did generate collections for my load order, so I have a collections json file. But after playing with collections a little, I turned them off until you add persistence.

**Comment by SteveTownsend** [comment]
Thanks - disabling Collections should work fine and restore the original behaviour. Containers and NPCs got some updates in this revision, I expect I broke something(s).

**Comment by jtolmer** [comment]
I believe that I narrowed down the issue. The Seeker "corpses" that aren't looting aren't NPCs. Rather, they're Activators. Weapon racks are also Activators. So even though I can click on the weapon in the rack and get information in the console, I'm assuming both are the same issue.

![20200704155131_1](https://user-images.githubusercontent.com/3989611/86522301-babe2880-be10-11ea-9f4c-57eb1727c004.jpg)

**Comment by SteveTownsend** [comment]
ACTI handling did change in this release. I'll take a look with that in mind, thanks v much for digging.

**Comment by SteveTownsend** [comment]
For the Ashfallow Citadel case - is this the rack?



If so, failure to loot could be the issue where Valuables are glowed even
if enchanted. I got an enchanted mace off this rack looted by changing *Items
that are worth more than this are treated as Valuable* to *Pick up*.

Choosing *Glow* as the default in v1.5 was a bad idea - I wrote it up as
below on the mod front page but it's really confusing. Is it possible this
is your issue? What about the Seeker case - which book was left behind?

Mod Features

v1.6 : release date July 3 2020. Hunterborn (WIP), World's Dawn integration
by wyongcan2019
<https://www.nexusmods.com/skyrimspecialedition/users/74244653>, options
for more realistic looting, bug fixes
v1.5 : Collections proof of concept - full workflow, but no persistence of
state across game reload - rebuilt on each reload from player starting
inventory, plus new items. Documentation in progress here
<https://www.nexusmods.com/skyrimspecialedition/articles/2232>﻿.
New defaults for *Valuables *are over-riding prior settings for
*Enchanted *items
- Open MCM and adjust the *Valuables *settings if you want to work round
this.

**Comment by jtolmer** [comment]
Yes, that's the weapon rack. I came across it while doing Served Cold, if the active quest matters.

I thought at first, before I responded about Activators, that the issue was related to Valuables as the mace I got was an enchanted elven one worth about 850. And then the first Seeker I noticed only had valuable books. But then the next seeker had an unread book that didn't get looted worth only 25 septims.
![20200704204952_1](https://user-images.githubusercontent.com/3989611/86525146-22d43500-be38-11ea-97f3-07c841ab9c0d.jpg)

Since 1.5 first released, I quickly changed the action for Valuables to Take All and have been running that way since. When I downgrade to 1.5, the items get picked up - they don't glow.

When trying to isolate it, I tried multiple tests changing around the Valuables settings to see if that was the issue. I was able to get the mace to glow by changing the action, but I never got the mace to be automatically picked up in 1.6 even when setting the action to Take All and / or moving the Item Value for valuables down to min or up to max.

In case the new-to-1.6 behavior of not looting past doors was involved, I also tried running with that setting both on and off. It didn't seem to matter.

**Comment by SteveTownsend** [comment]
I did not get back to this yet, I need to understand and fix this before
pushing out a revision but another change that is needed got more
complicated. This is in Location Tracking which underpins a lot of what I
want to add in v2++.

**Comment by SteveTownsend** [comment]
I tried the weapon rack again - everything loots OK for me, subject to my settings permitting. I am going to drop am update to 1.6.0.0 and then revisit the Seeker issue if it's still a problem.

**Comment by SteveTownsend** [comment]
v1.6.1.0 is up on Nexus. Please let me know if this fixes one or both of
your issues. There's enough change that it might, but no targeted fix
based on a repro of your reports, I am afraid.

**Comment by jtolmer** [comment]
I installed v1.6.1.0 and proceeded to the weapon rack. The mace didn't loot. I then went into MCM and loaded Smart Harvest defaults. It looted. Going through and comparing my settings to the defaults, I figured out that it seems to not loot for me due to having set "Global Value / Weight Default" to 25. But there's still a bug because the mace I have in the weapon rack has a V/W of 50. If I change the setting to the default of 10, then it loots.

https://youtu.be/K1PTWIxRKjA

That's the weapon rack. The Seekers are still broken for me in v1.6.1.0 with both my settings and the defaults.

https://youtu.be/VGL-JfGbg1Y

But if I downgrade to 1.5.0.1 and load the save, the Seeker gets looted using the my settings.

https://youtu.be/m-xeKOA6GJA

Hope that helps.

**Comment by SteveTownsend** [comment]
This is very helpful, I'll look later tonight and this week. I'm sorry it's
malfunctioning.

**Comment by jtolmer** [comment]
No worries about the behavior I'm running into. Using Smart Harvest is still a huge win for me even with these issues. Thank you for creating it and working to make it better. Your work is very much appreciated.

**Comment by SteveTownsend** [comment]
Thanks, that means a lot.

re the mace. I am suspicious that the value calculations I took as part of World's Dawn support broke this. I've pulled all of the item-value part of that code out and rebuilt - would you be able to give this v1.6.2.0 prerelease DLL a try? https://drive.google.com/file/d/1AUEhOLbgRyEDwC3tsVLSqGMw0D6IiqBE/view?usp=sharing

re the Seeker. Someone else has reported AshPiles failing to loot. I made changes recently to modify how Summons are handled, I wonder if Seekers and ash piles are mutually broken by that change?

**Comment by SteveTownsend** [comment]
I am pretty sure I flat out broke ashpiles, I'll update that DLL when I can finagle it back in

**Comment by SteveTownsend** [comment]
I updated the DLL with a fix for lootable ashpiles (Seekers spawn these when they die)

https://drive.google.com/file/d/1AUEhOLbgRyEDwC3tsVLSqGMw0D6IiqBE/view?usp=sharing

**Comment by jtolmer** [comment]
The latest DLL fixed the mace in the weapon rack for me. It did not seem to fix looting from the pile of rags that result from killing a a Seeker, though it does seems to work with the Ash Piles from Ash Spawn.

**Comment by SteveTownsend** [comment]
Finally test evidence for these cases, now I see how to test-spawn using console
Ice Wraith/Ice Pile:
```
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(121): [VERBOSE] {20700} (2020-07-07 06:46:48 AM) dynamic REFR 0xff001351 or base 0x00023ab3 for Ice Wraith
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(408): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) scanning container/body Ice Wraith/0xff001351  at distance 241.44 units
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(155): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) 00023ab3 2b(43) [Ice Wraith]
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(157): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) RE::TESContainer--
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(188): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) ExtraContainerChanges--
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(200): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) - 0005e99c [] 0000028D11EA1308 count=1 playable=0  - unknown
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(200): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) - 95041a5e [Dawnbreaker Hit] 0000028D11B84EE8 count=1 playable=0  - unknown
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(200): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) - 0003ad6a [Ice Wraith Teeth] 0000028D12019428 count=1 playable=1  - ingredient
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(129): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) Worth(40.00)  Weight(0.10)  V/W(400.00)
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(531): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) block looted container Ice Wraith/0xff001351
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(593): [VERBOSE] {20700} (2020-07-07 06:46:48 AM) Collectible Item 0x0003ad6a
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(639): [MESSAGE] {20700} (2020-07-07 06:46:48 AM) get Ice Wraith Teeth (1) from container Ice Wraith/0xff001351
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(738): [VERBOSE] {20700} (2020-07-07 06:46:48 AM) Trigger glow for Ice Wraith/0xff001351
J:\GitHub\SmartHarvestSE\src\Looting\InventoryItem.cpp(22): [VERBOSE] {20700} (2020-07-07 06:46:48 AM) get Ice Wraith Teeth/0x0003ad6a (1)
J:\GitHub\SmartHarvestSE\src\Looting\InventoryItem.cpp(78): [VERBOSE] {20700} (2020-07-07 06:46:48 AM) Move item Ice Wraith Teeth (1)
```

**Comment by SteveTownsend** [comment]
This appears to work for all the cases above, I wonder what else could be causing the Seeker problem? I'm going to make v1.6.2.0 based on the above testing but can keep digging on this.

**Comment by SteveTownsend** [comment]
H aha I actually saw failure to loot an Ice Wraith pile just after posting this. I tin I got to it before the script has populated the ACTI - I am trying the test again with a delay.

**Comment by SteveTownsend** [comment]
fixed in v1.6.2.0, regression as of v1.6.0.0

**Comment by jtolmer** [comment]
Confirmed that both my cases are fixed in 1.6.2. Thank you very much.

**Comment by SteveTownsend** [comment]
that's great, thanks for checking so quickly. The looting logic should not
change very much from now on, thankfully. Plenty of bells and whistles in
mind related to state of the world, though.
